src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let output = '';
+
+    for (const item of this) {
+      if (item !== null && item !== undefined) {
+        if (this.indexOf(item) !== this.length - 1) {
+          output += `${item}${separator}`;
+        } else {
+          output += item;
+        }
+      } else {
+        if (this.indexOf(item) !== this.length - 1) {
+          output += `${separator}`;
+        }
+      }
+    }
+
+    if (separator === '') {
+      return output;
+    }
```

Comment:
This code does nothing, `output` is returned anyway.

Reply:
I couldn't agree more. Thank you :)